pkg/cluster/px/px.go

```diff
@@ -1015,6 +1022,32 @@ func (ops *pxClusterOps) parseConfigFromDaemonset() (
 	return endpoints, opts, clusterName, platform, affinity, nil
 }
 
+func (ops *pxClusterOps) extractEnv(varName string, specEnv []v1.EnvVar, namespace string) string {
+	if len(varName) < 2 || varName[0] != '$' {
+		return varName
+	}
+	look4 := varName[1:] // skip '^$'
+	for _, v := range specEnv {
+		if v.Name != look4 {
+			continue
+		}
+		// if value defined, return immediately
+		if v.Value != "" {
+			return v.Value
+		}
+		// else.. let's dig through EnvSource
+		if v.ValueFrom != nil && v.ValueFrom.SecretKeyRef != nil {
+			val, err := ops.utils.GetSecret(v.ValueFrom.SecretKeyRef, namespace)
+			if err != nil {
+				logrus.WithError(err).Errorf("Could not extract %q secret", v.ValueFrom.SecretKeyRef)
+				return varName
```

Comment:
in cases where we cannot lookup the value of the env variable key, let's return err instead of returning the `varName` itself. Better to be more specific and return error instead fo returning the key itself since the key won't work for the caller who is expected it's value.

Reply:
Well.. this may not be a "clear cut" case:
* we are resolving `password:$ENV_VAR` reference:
   1. if we find `$ENV_VAR` listed as an `env{ ENV_VAR=foo123 }` -- we'll use `password:foo123`
   2. if we find it listed as `env{ valueFrom: secretKeyRef: ...}` -- we'll use `password:<deference-secret>`
   3. the "final" case -- we'll return `password:$ENV_VAR` (verbatim)

The 3rd case is perfectly legal when no references in `env{}` block were found -- but, like you said, it might make more sense to return an error if we were _supposed to_ extract the entry from a secret, and we couldn't.

Reply:
Note, these concerns should be addressed in the follow-up d06b2c0